metabase/metabase
# Replace ButtonGroup component with SegmentedControl

This PR replaces [`ButtonGroup` component](https://github.com/metabase/metabase/blob/v0.40.0/frontend/src/metabase/components/ButtonGroup.jsx) with the `SegmentedControl`. `ButtonGroup` is just a group of styled divs with click handlers, so it's not really accessible. On the other hand, `SegmentedControl` renders a list (`ul`) of `<input type="radio" />` with labels. No visual or functional changes are expected here, just a pure a11y fix.

### To Verify

1. Create a question displayed as a line chart
    * Ask a question > Custom Question > Sample Dataset > Products
    * Summarize _Average of Price_ by _Category_
    * Click the "Visualise" button in the bottom left corner
    * Click the "Visualization" button in the bottom left corner
    * Select "Line"
    * Click the "Settings" button in the bottom left corner
    * Select the "Display" option on the left sidebar
2. Ensure "Line style", "Show dots on lines" and "Which axis" controls are displayed as before. Try changing these options and make sure the visualization settings are applied

### Demo

**Before**
I highlighted the elements using the `ButtonGroup`

![CleanShot 2021-07-13 at 11 05 13@2x](https://user-images.githubusercontent.com/17258145/125426262-077b8dc2-57b8-4f19-b984-9f3f6ce9a2da.png)

**After**
No visual changes are expected

https://user-images.githubusercontent.com/17258145/125426286-ebbe55ca-358f-420f-887d-5ad958a4fe0c.mov
